torch/fx/passes/_tensorify_python_scalars.py

```diff
@@ -244,13 +260,72 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
                         )
 
                     node.replace_all_uses_with(replacement_proxy.node)
-
                     graph.erase_node(node)
+                    deleted.add(node)
 
     # DCE symbols (which are guaranteed to be pure) only
-    for proxy in reversed(expr_to_sym_proxy.values()):
-        if len(proxy.node.users) == 0 and proxy.node.op != "placeholder":
-            graph.erase_node(proxy.node)
+    dce(graph, deleted)
+
+    # Now do one more pass that specializes all symfloats we didn't manage
+    # to tensorify away.
+    nodes = list(graph.nodes)
```

Comment:
from docs:

> Note that this Node list representation is a doubly-linked list. Mutations during iteration (e.g. delete a Node, add a Node) are safe.

Reply:
So it shouldn't be necessary to make a copy of the node list

Reply:
I need it for `graph.inserting_before` call

```
        with graph.inserting_before(
            nodes[i + 1] if node not in placeholders else first_non_placeholder
        ):
```

Or perhaps I'm misunderstanding. Can you propose a suggested change if so?

Reply:
ok appeal to preexisting condition accepted lol